### PR TITLE
Create magic mark initial design and login form

### PR DESCRIPTION
**Pull request by AnastasiyaPoleshuk**

### CODE REVIEW

File: magic-marks/src/api/loginAPI.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import CONSTANTS from '../utils/constants';
+import { ILoginFormData } from '../types/interfaces';
+
+const createUser = async (data: ILoginFormData) => {
```
Comment (AyvengoBY) [inline magic-marks/src/api/loginAPI.ts]: Rename to LoginUser, please.

File: magic-marks/src/components/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+          <input
+            type="password"
+            placeholder="Пароль"
+            className={`login-form__input ${errors.password ? 'input-error' : null}`}
+            {...register('password', { required: true, minLength: 3 })}
+          />
+
+          <p className={`form-error ${errors.password ? 'open' : null}`}>
+            *Длина пароля должна быть не меньше 3 символов
```
Comment (AyvengoBY) [inline magic-marks/src/components/LoginForm/LoginForm.tsx]: Please move all text values  into constants. For this purpose, please create locales/ru-Ru/index.json  file with localized string constunts (this file will be used for possible localization purpose using i18 package)

File: magic-marks/src/views/main/Main.tsx [hunk 3]
```diff
@@ -1,15 +1,54 @@
...
+          <div className="title title__part-one">
+            Magic
+            <span>Контролируй свою успеваемость</span>
+          </div>
+          <div className="title title__part-two">Marks</div>
+        </div>
+        <div className="description-block">
+          <div className="description-block__wrapp">
+            <p className="description description-bottom">
+              Приложение позволяет просматривать оценки,
```
Comment (AyvengoBY) [inline magic-marks/src/views/main/Main.tsx]: Move into constants, please

File: magic-marks/src/views/main/Main.tsx [hunk 4]
```diff
@@ -1,15 +1,54 @@
...
+              loginUser.isAuth
+                ? <NavLink to="/subjects" className="subjects__btn">Дневник</NavLink>
+                : <button type="button" className="log-in__btn" onClick={toggleModal}>Войти</button>
+            }
+          </div>
+          <img src={path} alt="main img" className="main-img" />
         </div>
-        <div className="main-img" />
       </div>
+      {isOpenModal
```
Comment (AyvengoBY) [inline magic-marks/src/views/main/Main.tsx:44]: Is this code can support different modals ? (Or Login modal only)

**Review by AyvengoBY — CHANGES_REQUESTED** [review]

File: magic-marks/src/api/loginAPI.ts (hunk 1 above)
Comment (AnastasiyaPoleshuk) [reply magic-marks/src/api/loginAPI.ts]: changed

File: magic-marks/src/components/LoginForm/LoginForm.tsx (hunk 2 above)
Comment (AnastasiyaPoleshuk) [reply magic-marks/src/components/LoginForm/LoginForm.tsx]: done

File: magic-marks/src/views/main/Main.tsx (hunk 3 above)
Comment (AnastasiyaPoleshuk) [reply magic-marks/src/views/main/Main.tsx]: done

File: magic-marks/src/views/main/Main.tsx (hunk 4 above)
Comment (AnastasiyaPoleshuk) [reply magic-marks/src/views/main/Main.tsx:44]: yes

**Review by AyvengoBY — APPROVED** [review]

**Pull request merged**